InfluxDB: Add delete buttons to measurement, part and tag in query editor

AlexGodbehere opened the pull request:
**What is this feature?**

This PR implements a delete button to the `measurement`, `part` and `tag` sections in the InfluxDB datasource editor. This improves user experience by allowing users to delete measurements, parts and tags directly from the user interface. The commit also slightly refactors the options generation for the `where` clause, removing the `-- remove filter --` option as it becomes redundant with the new delete button.

Before:
<img width="582" alt="image" src="https://github.com/grafana/grafana/assets/114239316/730c169b-a54e-4aeb-abfe-c51f7c4513b5">

After:
<img width="660" alt="image" src="https://github.com/grafana/grafana/assets/114239316/69ff3445-05df-4317-a56c-d1b6936cbccb">


**Why do we need this feature?**

Before this change there was no way to remove `GROUP BY` entries or clear the currently selected measurement, which caused an issue because available tag values were filtered by the current measurement. The only solution was to remove the query and start again or splice out the offending code in the raw query editor.

**Who is this feature for?**

Users of the InfluxDB datasource that use the InfuxQL graphical query editor.

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

AlexGodbehere committed:
InfluxDB/QueryEditor: Add delete buttons to measurement, part and tag sections

The diff provides an additional "delete" button to measurement section, part list section and tags section in the influxdb datasource editor. This improves user experience by allowing users to delete measurements, parts and tags directly from the user interface. The commit also slightly refactors the options generation for these sections, removing the "-- remove filter --" option as it becomes redundant with the new delete button.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=72825) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=72825) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=72825) it.</sub>

catherineymgui commented:
Thanks for your contribution!

It would be great if we could match some of the other data sources (Prometheus is the one we try to aspire to). Would you be able to use the "X" instead of the trash icon?

<img width="548" alt="Screenshot 2023-08-04 at 08 50 14" src="https://github.com/grafana/grafana/assets/5264102/daef1610-bd60-4e5d-b760-5eeed2a11746">

AlexGodbehere committed:
InfluxDB/QueryEditor: Replace Button with AccessoryButton

Changed the Button components to AccessoryButton from the '@grafana/experimental' package, used for the delete functionality in data source components for influxdb query editor. The AccessoryButton is more visually consistent with our current Grafana UI aesthetic and provides better user experience. The change was implemented in FromSection.tsx, PartListSection.tsx, and TagsSection.tsx files.

AlexGodbehere commented:
Sure thing. I've moved over to `AccessoryButton` as Prometheus uses. It doesn't look as swish as Prometheus but it's a step closer.

<img width="617" alt="image" src="https://github.com/grafana/grafana/assets/114239316/0dea3bbb-f8dc-40c4-b11f-bfc1f3c35460">

AlexGodbehere committed:
Merge branch 'main' into main

AlexGodbehere commented:
@catherineymgui are you a reviwer for this? What's the process for getting this approved?

itsmylife commented on public/app/plugins/datasource/influxdb/components/editor/query/influxql/visual/FromSection.tsx:
```diff
@@ -1,6 +1,9 @@
+import { AccessoryButton } from '@grafana/experimental';
+
 import { DEFAULT_POLICY } from '../../../../../types';
+
```
```suggestion
```

itsmylife replied:
Just to make `eslint` happy

itsmylife commented on public/app/plugins/datasource/influxdb/components/editor/query/influxql/visual/PartListSection.tsx:
```diff
@@ -125,17 +126,28 @@ export const PartListSection = ({
-            onChange(index, pars);
-          }}
-        />
+        <>
+          <Part
+            key={index}
+            name={part.name}
+            params={part.params}
+            onRemove={() => {
+              onRemovePart(index);
+            }}
+            onChange={(pars) => {
+              onChange(index, pars);
+            }}
+          />
+          <AccessoryButton
+            style={{ marginRight: '4px' }}
+            aria-label="remove"
+            icon="times"
+            variant="secondary"
+            onClick={() => {
+              onRemovePart(index);
+            }}
+          />
+        </>
```
```suggestion
        <React.Fragment key={index}>
          <Part
            name={part.name}
            params={part.params}
            onRemove={() => {
              onRemovePart(index);
            }}
            onChange={(pars) => {
              onChange(index, pars);
            }}
          />
          <AccessoryButton
            style={{ marginRight: '4px' }}
            aria-label="remove"
            icon="times"
            variant="secondary"
            onClick={() => {
              onRemovePart(index);
            }}
          />
        </React.Fragment>
```

itsmylife replied:
Moving the `key` to `Fragment`

AlexGodbehere committed:
Update public/app/plugins/datasource/influxdb/components/editor/query/influxql/visual/FromSection.tsx

Co-authored-by: ismail simsek <ismailsimsek09@gmail.com>

AlexGodbehere committed:
Update public/app/plugins/datasource/influxdb/components/editor/query/influxql/visual/PartListSection.tsx

Co-authored-by: ismail simsek <ismailsimsek09@gmail.com>

itsmylife reviewed (approved):
LGTM 🚀 
Thanks for the contribution!
